src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -119,27 +124,30 @@ class LibraryService (protected val argUserInfo: UserInfo, val rawlsDAO: RawlsDA
   }
 
   def publishDocument(ws: Workspace): Unit = {
-    searchDAO.indexDocument(indexableDocument(ws))
+    indexableDocument(ws, ontologyDAO) onSuccess {
+      case document => searchDAO.indexDocument(document)
+    }
   }
 
   def removeDocument(ws: Workspace): Unit = {
     searchDAO.deleteDocument(ws.workspaceId)
   }
 
   def indexAll: Future[PerRequestMessage] = {
-    rawlsDAO.getAllLibraryPublishedWorkspaces map {workspaces =>
+    rawlsDAO.getAllLibraryPublishedWorkspaces flatMap { workspaces: Seq[Workspace] =>
       searchDAO.recreateIndex()
       if (workspaces.isEmpty)
-        RequestComplete(NoContent)
+        Future(RequestComplete(NoContent))
       else {
-        val toIndex:Seq[Document] = workspaces.map {workspace => indexableDocument(workspace)}
-        val bi = searchDAO.bulkIndex(toIndex)
-        val statusCode = if (bi.hasFailures) {
-          InternalServerError
-        } else {
-          OK
+        val toIndex: Future[Seq[Document]] = Future.sequence(workspaces map { indexableDocument(_, ontologyDAO) })
+        toIndex map { documents =>
+          val indexed = searchDAO.bulkIndex(documents)
```

Comment:
nitpick - the variable name looks like a boolean but it's a full `LibraryBulkIndexResponse`. One could also iterate through the failures and dig out the individual errors, but that seems like overkill since the whole `indexed` is being returned as the error body.